### PR TITLE
Add cache tags to PriceDefaultFormatter.

### PR DESCRIPTION
**Pull request by facine**
We need to expire the cache if change the formatter settings.

### CODE REVIEW

File: modules/price/src/Plugin/Field/FieldFormatter/PriceDefaultFormatter.php [hunk 1]
```diff
@@ -142,6 +142,9 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
             'languages:' . LanguageInterface::TYPE_INTERFACE,
             'country',
           ],
+          'tags' => [
+            'config:' . $items[$delta]->getFieldDefinition()->getTargetEntityTypeId() . '.' . $items[$delta]->getFieldDefinition()->getTargetBundle() . '.' . $this->viewMode,
```
Comment (bojanz) [inline modules/price/src/Plugin/Field/FieldFormatter/PriceDefaultFormatter.php]: $this->fieldDefinition should be the same as $items[$delta]->getFieldDefinition(), right? That one is shorter.
Looks like this tag doesn't change between deltas, which means we can calculate it above the loop (define a $cache_tag above $elements = [];)

**Comment by facine** [comment]
You're in the right! Fixed!

Thanks!

**Comment by swentel** [comment]
When changing entity displays, everything with 'rendered' is cleared anyway, so you should be fine. Most default formatters in core don't have any special cache tags, except then for files/image formatters maybe. So unless you're doing something funky, I think you're fine with the current formatters.

(And even if we'd need a cache tag on the field, then we should rather add this to core)

**Comment by facine** [comment]
By anyway when I try this locally don't work, now works fine without this patch...

Sorry!

**Comment by Berdir** [comment]
Yeah. this shouldn't be needed. the cache tag also doesn't seem valid, it is missing the prefix  "core.entity_view_display.", this doesn't actually do something right now.

**Pull request closed without merging**